connect/api/src/main/java/org/apache/kafka/connect/data/Struct.java

```diff
@@ -229,7 +229,11 @@ public void validate() {
             Object value = values[field.index()];
             if (value == null && (fieldSchema.isOptional() || fieldSchema.defaultValue() != null))
                 continue;
-            ConnectSchema.validateValue(fieldSchema, value);
+            try {
+                ConnectSchema.validateValue(fieldSchema, value);
+            } catch(DataException e) {
+                throw new DataException("Invalid value: null used for required field: "  + field.name() + ", schema type: " + fieldSchema.type());
```

Comment:
While current implementation of `validateValue` only throws DataException when a required field has null value, in general we may have other exception in the future, right? Then it may be misleading to claim that "null used for required field" here. Maybe you can do `throw new DataException(..., e)` here.